### PR TITLE
Implement QuickPass, logging and a few other goodies

### PR DESCRIPTION
**Pull request by alexhauser**
#37 , #13 

### Description:
This is a rather large update that implements:

- The _QuickPass_ feature (including receiving and handling system events for clearing QuickPass entries (currently Windows only))
- Logging using the `Serilog` package
- Taskbar notification area icon (aka "tray icon") on Windows

### Remarks:

_QuickPass_ requires the app to keep running in the background (_QuickPass_ information may only ever be stored in RAM). So I had to make some changes:

- On Windows, I implemented a tray icon. Closing the main window will hide the window instead of exiting the app. Double-clicking the tray icon will bring it back.
- On Linux and Mac, for now the app just minimizes to the dock/panel, whatever. I hope we can implement something nice there, too.

**Logging**:
Serilog is currently configured to log to the console as well as to a file (daily rotation). We can change the log sinks at any time of course.

**Platform-specific implementations**:
Since we are now entering territory where platform-specific implementations are necessary (tray icon, system event notifications for clearing _QuickPass_), I have made an attempt to abstract the ugly part of all of this away by:

- Creating an Interface for each feature that needs platform-specific implemenations
(e.g `INotifyIcon` for the tray icon). The application code will only ever work with those interfaces, without needing to know about the actual implementation.
- Creating a class `Implementation` with a static function `ForType<T>()` that acts as a single place for registering platform-specific implementations for the above mentioned interface types.
Here is what the calling code looks like for the tray icon:
```C#
            // Set up and configure the notification icon

            // Get the type of the platform-specific implementation
            Type type = Implementation.ForType<INotifyIcon>();
            if (type != null)
            {
                // If an implementation exists for the current platform, create an instance for it
                NotifyIcon = (INotifyIcon)Activator.CreateInstance(type);
            }

            // The code that follows will always check if we actually have an instance
            if (NotifyIcon != null)
            {
                ....
            }
```

### CODE REVIEW

**Comment by josegomez** [comment]
Damn! this is down right beatiful @alexhauser  checking it out now great job

File: SQRLDotNetClientUI/Models/INotifyIcon.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿using Avalonia.Controls;
+using System;
+
+namespace SQRLDotNetClientUI.Models
```
Comment (josegomez) [inline SQRLDotNetClientUI/Models/INotifyIcon.cs:4]: Very nice I love the interface / abstraction idea!

Comment (alexhauser) [reply SQRLDotNetClientUI/Models/INotifyIcon.cs:4]: Cool, glad you like it! 😁

**Review by josegomez — APPROVED** [review]
Awesome work! Merge Away!

**Pull request merged**